src/lib/draw/types.ts

```diff
@@ -0,0 +1,26 @@
+import type { Feature, Geometry } from "geojson";
+
+export type Mode =
+  | {
+      mode: "list";
+    }
+  | {
+      mode: "edit-form";
+      id: number;
+    }
+  | {
+      mode: "edit-geometry";
+      id: number;
+    }
+  | { mode: "new-point" }
+  | { mode: "new-freehand-polygon" }
+  | { mode: "new-snapped-polygon" }
+  | { mode: "new-route" }
+  | { mode: "split-route" }
+  | { mode: "set-image" }
+  | { mode: "streetview" };
+
+// Properties are guaranteed to exist, but can be anything, so TS is mostly disabled
```

Comment:
I renamed this to be more clear. By setting the properties this way, TS assumes any key can exist, with any type. (It's what we've been doing before; there's no change, just the name makes it more clear)